### DISCUSSION TITLE
SignalR client.start() function blocked after keepalive timeout error

### DISCUSSION THREAD

**Discussion by supremeremixx** [theOehrly/Fast-F1#627] [Q&A] [answered]
Hello!
I'm trying to run endless loop with SignalRClient.

```
import time
import logging
from fastf1.livetiming.client import SignalRClient

log = logging.getLogger()
log.setLevel(logging.DEBUG)

def signalr():
  client = SignalRClient(filename="output.txt", timeout=1800, debug=True)
  client.start()

while True:
  signalr()
  time.sleep(60)
```

After random amount of time (3, 5 , 11 hours...) I get keepalive timeout error. In logs I see that TCP connection is closed, but function client.start() still blocked, it never ends. Loop is never finished and that's why program not creating a new client instance after error.

```
python3[282709]: 2024-08-09 21:36:44,532 - DEBUG: ! timed out waiting for keepalive pong
python3[282709]: 2024-08-09 21:36:44,533 - DEBUG: ! failing connection with code 1011
python3[282709]: 2024-08-09 21:36:44,535 - DEBUG: = connection is CLOSING
python3[282709]: 2024-08-09 21:36:44,535 - DEBUG: > CLOSE 1011 (internal error) keepalive ping timeout [24 bytes]
python3[282709]: 2024-08-09 21:36:54,536 - DEBUG: ! timed out waiting for TCP close
python3[282709]: 2024-08-09 21:36:54,536 - DEBUG: x closing TCP connection
python3[282709]: 2024-08-09 21:37:04,539 - DEBUG: ! timed out waiting for TCP close
python3[282709]: 2024-08-09 21:37:04,539 - DEBUG: x aborting TCP connection
python3[282709]: 2024-08-09 21:37:04,540 - DEBUG: = connection is CLOSED
python3[282709]: 2024-08-09 21:59:44,619 - WARNING: Timeout - received no data for more than 1800 seconds!
```

I've tried running it with and without timeout (in SignalRClient), but the function remains locked.
Is it possible to somehow terminate it after receiving an error?

**Comment by theOehrly** [comment]
Hi, I'm really not sure what causes this and how to fix it at this stage. I've never seen this error myself, and it's the first time it has been reported.
I agree that ``.start()`` should not remain blocking but instead return. But I am not sure how soon I'll realistically get around to fixing this.

**Reply by theOehrly** [reply, marked as answer]
I've opened #628 to track this as an issue.

**Reply by supremeremixx** [reply]
Thanks! Appreciate it!
